rust/src/lib/route.rs

```diff
@@ -77,6 +79,34 @@ impl Routes {
                     error!("{}", e);
                     return Err(e);
                 }
+                if let Some(dst) = route.destination.as_ref() {
+                    let dst_str = dst.as_str();
+                    if !is_ipv6_addr(dst_str) {
+                        let mut is_invalid_dst = false;
+                        if dst_str == "0.0.0.0" {
+                            is_invalid_dst = true;
+                        } else if dst_str.contains('/') {
+                            let ip_net: Vec<&str> =
+                                dst_str.split('/').collect();
+                            let ip_addr = Ipv4Addr::from_str(ip_net[0]);
+                            let prefix = ip_net[1].parse::<i32>().unwrap_or(32);
```

Comment:
The `unwrap_or(32)` is wrong. What if I use `0.0.0.0/a`.

Reply:
BTW. The `ip_net[1]` will panic if the dst is `1.1.1.1`.

Reply:
Using `[]` is dangerous unless you really sure the index exists. I think `ip_net[0]` might cause problem on empty string `dst`.

Reply:
> BTW. The `ip_net[1]` will panic if the dst is `1.1.1.1`.

I think that the dst will not be `1.1.1.1`, because we the code will be called under the if condition `else if dst_str.contains('/') `.

Reply:
> Using `[]` is dangerous unless you really sure the index exists. I think `ip_net[0]` might cause problem on empty string `dst`, as a library, we should **never** panic!

I am pretty sure that the index exists base on the if condition `else if dst_str.contains('/')`. The dst can not be empty string under the if condition.